### PR TITLE
Use promise instead Callback in requestPermission

### PR DESCRIPTION
**Pull request by davinkevin**
Implementation link to the ticket https://github.com/neoziro/angular-notification/issues/4

### CODE REVIEW

**Comment by davinkevin** [comment]
It failed on the npm install, don't seem to be linked to the PR...?

File: test/angular-notification.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ describe('Notification provider', function () {
   beforeEach(inject(function ($injector) {
     $window = $injector.get('$window');
     Notification = $injector.get('Notification');
+    $q = $injector.get('$q');
```
Comment (gregberge) [inline test/angular-notification.js]: Unused.

File: examples/simple-example.html [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
+                    var notification = new Notification('New message', {
+                        body: 'You have a new message.',
+                        delay: 2000
+                    });
 
-        notification.$on('click', function () {
-          console.log('The user has clicked in my notification.');
-        });
+                    notification.$on('show', function () {
+                        console.log('My notification is displayed.');
```
Comment (gregberge) [inline examples/simple-example.html]: Please use $log, I have just seen it but it's preferable to use it.

File: angular-notification.js [hunk 3]
```diff
@@ -121,18 +123,21 @@ function NotificationProvider() {
...
      * Static method to request permission.
-     *
-     * @param {Function} callback
+     * It returns a promise
      */
 
-    NgNotification.requestPermission = function (callback) {
-      if (! $window.Notification) return false;
+    NgNotification.requestPermission = function () {
+        var deferred = $q.defer();
```
Comment (gregberge) [inline angular-notification.js]: @davinkevin I think it's preferable to use the promise like implementation than deffered:

``` js
return $q(function (resolve, reject) { ... });
```

**Comment by gregberge** [comment]
@davinkevin commented

**Comment by gregberge** [comment]
@davinkevin please try to upgrade angular to '^1.3.14' in bower.json.

**Comment by gregberge** [comment]
@davinkevin OK just rebase on master please, I fixed tests. And after I will merge it.

**Comment by davinkevin** [comment]
I have merged with the original master. it is alright ? 
If you need more modification, let me know, I'm new with PR on github, so if you have some instruction to help me to do the work well, I'll be glad to see it. 

Thanks

**Comment by gregberge** [comment]
@davinkevin you can try to document on rebase process, but merge is OK. Thank you.

**Pull request merged**